### PR TITLE
Add Consent Mode settings infrastructure.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #8270 

## Relevant Technical Choices

- I've only added tests for the logic, not the boilerplate.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Plugin.php [hunk 1]
```diff
@@ -212,7 +212,7 @@ function() use ( $options, $activation_flag ) {
 				( new Core\Util\Migration_1_8_1( $this->context, $options, $user_options, $authentication ) )->register();
 				( new Core\Dashboard_Sharing\Dashboard_Sharing( $this->context, $user_options ) )->register();
 				( new Core\Key_Metrics\Key_Metrics( $this->context, $user_options, $options ) )->register();
-				( new Core\Prompts\Prompts( $this->context, $user_options ) )->register();
```
Comment (aaemnnosttv) [inline includes/Plugin.php:215]: This looks like it wasn't intended to be removed/replaced.

File: tests/phpunit/integration/Core/Consent_Mode/Consent_Mode_SettingsTest.php [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+ *
+ * @package   Google\Site_Kit\Tests\Core\Consent_Mode
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Modules;
+
+use Google\Service\Compute\Region;
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/Core/Consent_Mode/Consent_Mode_SettingsTest.php:13]: Accidental import?

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @techanvil – a few observations but overall looks great 👍

File: includes/Plugin.php (hunk 1 above)
Comment (techanvil) [reply includes/Plugin.php:215]: Whooop... move fast and break things, right? :sweat_smile: 

Thanks for the spot. This is now fixed.

File: tests/phpunit/integration/Core/Consent_Mode/Consent_Mode_SettingsTest.php (hunk 2 above)
Comment (techanvil) [reply tests/phpunit/integration/Core/Consent_Mode/Consent_Mode_SettingsTest.php]: Thanks, this is now removed.

**Review by aaemnnosttv — APPROVED** [review]
Great, thanks @techanvil 🚀

**Pull request merged**